test/agentchat/test_groupchat.py

```diff
@@ -1007,6 +1006,188 @@ def chat(gc_manager: autogen.GroupChatManager):
+    The speaker_selection_method is defined to return the next agent in the expected sequence.
+    """
+
+    # For loop that creates UserProxyAgent with names from a to z
+    agents = [
+        autogen.UserProxyAgent(
+            name=chr(97 + i),
+            default_auto_reply=f"My name is {chr(97 + i)}",
+            human_input_mode="NEVER",
+            code_execution_config={},
+        )
+        for i in range(26)
+    ]
+
+    # Initiate allowed speaker transitions
+    allowed_or_disallowed_speaker_transitions = {}
+
+    # Each agent can transition to the next alphabet as a baseline
+    # Key is Agent, value is a list of Agents that the key Agent can transition to
+    for i in range(25):
+        allowed_or_disallowed_speaker_transitions[agents[i]] = [agents[i + 1]]
+
+    # The test is to make sure that the agent sequence is a,u,t,o,g,e,n, so we need to add those transitions
+    expected_sequence = ["a", "u", "t", "o", "g", "e", "n"]
+    current_agent = None
+    previous_agent = None
+
+    for char in expected_sequence:
+        # convert char to i so that we can use chr(97+i)
+        current_agent = agents[ord(char) - 97]
+        if previous_agent is not None:
+            # Add transition
+            allowed_or_disallowed_speaker_transitions[previous_agent].append(current_agent)
+        previous_agent = current_agent
+
+    def custom_speaker_selection_func(last_speaker: Agent, groupchat: GroupChat) -> Optional[Agent]:
+        """
+        Define a customized speaker selection function.
+        """
+        expected_sequence = ["a", "u", "t", "o", "g", "e", "n"]
+
+        last_speaker_char = last_speaker.name
+        # Find the index of last_speaker_char in the expected_sequence
+        last_speaker_index = expected_sequence.index(last_speaker_char)
+        # Return the next agent in the expected sequence
+        if last_speaker_index == len(expected_sequence) - 1:
+            return None  # terminate the conversation
+        else:
+            next_agent = agents[ord(expected_sequence[last_speaker_index + 1]) - 97]
+            return next_agent
+
+    groupchat = autogen.GroupChat(
+        agents=agents,
+        messages=[],
+        max_round=20,
+        speaker_selection_method=custom_speaker_selection_func,
+        allowed_or_disallowed_speaker_transitions=allowed_or_disallowed_speaker_transitions,
+        speaker_transitions_type="allowed",
+    )
+    manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=False)
+
+    results = agents[0].initiate_chat(manager, message="My name is a")
+    actual_sequence = []
+
+    # Append to actual_sequence using results.chat_history[idx]['content'][-1]
+    for idx in range(len(results.chat_history)):
+        actual_sequence.append(results.chat_history[idx]["content"][-1])  # append the last character of the content
+
+    assert expected_sequence == actual_sequence
+
+
+def test_custom_speaker_selection_contradicts_transition_graph():
```

Comment:
Hey @kevin666aa, not sure if you are aware, if the suggested agent from the custom select function is not a legal transition in the transition graph, the current logic overrides the constraints of the graph (1). It could be alright, just checking that this is intentional, or if you also wanted to raise a warning/ error (2).

Reply:
Hello @joshkyh, yes I think it would be good to return a warning, but not to raise an error directly? 
Both the custom function and the graph transition are defined by user.
Also we pass in the groupchat object to custom function. We would expect user to utilize the defined transitions in the groupchat object, and not passing in conflict parameters for graph transition and custom function.

If we are to change this, I think we should open a separate PR.

Reply:
It sounds reasonable to me! I'll change the test name to custom function overrides graph transitions tonight.